sleuth_backend/solr/connection.py

```diff
@@ -52,18 +58,49 @@ def fetch_core_schema(self, name):
         return response['schema']
 
     def insert_document(self, core, doc):
-        """
-        Attempts to insert the given document (a dict) into the solr core with
-        the given name and returns the response from Solr. All values in 'doc'
-        must be strings.
-        """
+        '''
+        Queues a document for insertion into the specified core and returns None.
+        If the number of documents in the queue exceeds a certain threshold,
+        this function will insert them all the documents held in the queue of the
+        specified core and return the response from Solr.
+        All values in 'doc' must be strings.
+        '''
         if core not in self.cores:
             raise ValueError("A core for the document type {} was not found".format(core))
-        return self.cores[core].add([doc])
+
+        self.queues[core].append(doc)
+
+        if len(self.queues[core]) >= self.QUEUE_THRESHOLD:
+            docs = list(self.queues[core].copy())
+            del self.queues[core][:]
+            return self.insert_documents(core, docs)
```

Comment:
Given that there could be quite a few large objects in the queue it might be better to do
```python

result = self.insert_documents(core, self.queues[core])
del self.queues[core][:]
return result
```
That way we don't have to copy a bunch of web pages sitting in memory.

Reply:
Hm yeah I think i added this because I was occasionally seeing two documents triggering the same queue to be inserted - can't seem to be able to reproduce anymore tho, I'll get rid of it for now

Reply:
@bfbachmann actually removing the copy is causing some super whack behaviour, especially with the tests, making it impossible to verify what gets inserted... might leave the `copy()` in for now @_@